src/Microsoft.Azure.CosmosRepository/Repositories/DefaultRepository.Paging.cs

```diff
@@ -3,29 +3,47 @@
 
 
 // ReSharper disable once CheckNamespace
+
 namespace Microsoft.Azure.CosmosRepository;
 
 internal sealed partial class DefaultRepository<TItem>
 {
     /// <inheritdoc/>
-    public async ValueTask<IPage<TItem>> PageAsync(
+    public ValueTask<IPage<TItem>> PageAsync(
         Expression<Func<TItem, bool>>? predicate = null,
         int pageSize = 25,
         string? continuationToken = null,
         bool returnTotal = false,
-        CancellationToken cancellationToken = default)
-    {
+        CancellationToken cancellationToken = default) =>
+            PageAsync(
+                new QueryRequestOptions { MaxItemCount = pageSize },
+                predicate,
+                pageSize,
+                continuationToken,
+                returnTotal,
+                cancellationToken);
+
+    /// <inheritdoc/>
+    public async ValueTask<IPage<TItem>> PageAsync(
+        QueryRequestOptions requestOptions,
+        Expression<Func<TItem, bool>>? predicate = null,
+        int pageSize = 25,
+        string? continuationToken = null,
+        bool returnTotal = false,
+        CancellationToken cancellationToken = default)
+    {
         Container container = await containerProvider.GetContainerAsync()
-            .ConfigureAwait(false);
-
-        QueryRequestOptions options = new()
-        {
-            MaxItemCount = pageSize
-        };
+            .ConfigureAwait(false);
+
+        // make sure that if the user hasn't said the value already we take it from the pageSize parameter
+        if (requestOptions.MaxItemCount is null)
```

Comment:
You must expect that null could be passed. Initialize requestOption if the value is null.

Reply:
Or throw exception

Reply:
But the `requestOptions` isn't nullable, are you implying that we should protect against it anyway?

Reply:
By default, I don't trust users 😂 But if for you it's OK then for me too 😄